analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/ConditionEvaluatesToConstantBase.cs

```diff
@@ -30,15 +31,26 @@ public abstract class ConditionEvaluatesToConstantBase : SymbolicRuleCheck
     protected const string MessageFormat = "{0}";
     private const string MessageBool = "Change this condition so that it does not always evaluate to '{0}'.";
     private const string MessageNullCoalescing = "Change this expression which always evaluates to the same result.";
+    private const string MessageUnreachable = "{0} Some code paths are unreachable.";
```

Comment:
These are all messages, and they have different usages. I also find the `ReportIssue` below quite confusing because of that (see the comment there, as a result).

Let's make the primary message like this, where `{1}` will serve for optional suffix.
```
protected const string MessageFormat = "{0}{1}";
```
`MessageBool` remains.
`MessageNullCoalescing` remains - I'd name it just `MessageNull`, because it's valid for all null checking scenarios.
And then
```
private const string S2583MessageSuffix = " Some code paths are unreachable.";
```
This will allow renaming `issueMessage` simply to `message` or `prefix`. And to invoke `ReportIssue` without additional nested `string.Format`